### ISSUE TITLE
StringIndexOutOfBoundsException in ErrorFormatter 

### ISSUE THREAD

**Issue by thesamet** [sirthias/parboiled2#138]
I am getting this exception thrown from inside ErrorFormatter. Here is a minimal example that reproduces this behavior:

```scala
import org.parboiled2._
import scala.util._
import java.math.BigInteger

class TestParser(val input: ParserInput) extends Parser {
  def InputLine = rule {
    atomic(capture(oneOrMore(anyOf("ab")))) ~> {
      v: String => test(v.startsWith("a"))
    }
  }
}

object Hello extends App {
  val parser = new TestParser("babab")
  parser.InputLine.run() match {
    case Success(x) => println(x)
    case Failure(e: ParseError) => sys.error(parser.formatError(e, new ErrorFormatter(showTraces = true)))
    case x => println(x)
  }
}
```

The exception looks like this:
```
java.lang.StringIndexOutOfBoundsException: String index out of range: 6
	at java.lang.String.substring(String.java:1907)
	at org.parboiled2.ParserInput$StringBasedParserInput.sliceString(ParserInput.scala:98)
	at org.parboiled2.ErrorFormatter.formatProblem(ErrorFormatter.scala:80)
	at org.parboiled2.ErrorFormatter.format(ErrorFormatter.scala:58)
	at org.parboiled2.ErrorFormatter.format(ErrorFormatter.scala:51)
	at org.parboiled2.Parser.formatError(Parser.scala:105)
	at Hello$delayedInit$body.apply(Main.scala:17)
	at scala.Function0$class.apply$mcV$sp(Function0.scala:40)
```

Removing the `atomic` makes the exception go away, but I would like to use it for better error reporting.

**Comment by sirthias** [comment]
Thanks for this report, Nadav!
I'll get this fixed ASAP.

**Comment by dragostis** [comment]
I'm having a very similar issue. Fixing this would be extremely appreciated. :smile: 

And here's a :+1: because a star is not enough for how awesome this plugin is, even if it's can be a little rough around the edges right now.

**Comment by yanns** [comment]
same problem here

**Comment by sirthias** [comment]
Closed by 7a48ce2a44f0c13bca74b16fc511dca976e5aef2.
